tests/spatial/test_fss.py

```diff
@@ -0,0 +1,263 @@
+"""
```

Comment:
Can you add a test where `fcst` and `obs` have different dims (e.g., fcst has a lead day dim) to ensure that any broadcasting is done correctly?

Reply:
- I'll add a test for `NaN`s

- `dask` is actually out of scope for this PR. I've put warnings in the docstring, the option is there in the API but mainly "experimental/use at your own risk" for those who want a way to try it. Testing it at a unit test level only makes sense to me if there's an overarching integration test that already tests it's feasibility. I'd prefer to take out the `dask` option rather than testing it at this stage, if you're not keen on just having the warning.

- tests where `fcst` and `obs` have different dimensions are already included implicitly on most tests (via parametrization) that perform aggregation in which case fcst has `lead_time`. But I can make an explicit test if you prefer.

Reply:
Okay that all sounds good. No need for an explicit test for the last point.